core/pubnub_coreapi_ex.c

```diff
@@ -241,3 +242,60 @@ enum pubnub_res pubnub_history_ex(pubnub_t*                     pb,
     pubnub_mutex_unlock(pb->monitor);
     return rslt;
 }
+
+struct pubnub_set_state_options pubnub_set_state_defopts(void)
+{
+    struct pubnub_set_state_options options;
+
+    options.user_id = NULL;
+    options.channel_group = NULL;
+    options.heartbeat = false;
+
+    return options;
+}
+
+static enum pubnub_res heartbeat_with_state(pubnub_t *pb,
+        const char *channel,
+        const char *state,
+        struct pubnub_set_state_options opts)
+{
+    enum pubnub_res rslt;
+
+    PUBNUB_ASSERT(pb_valid_ctx_ptr(pb));
+
+    pubnub_mutex_lock(pb->monitor);
+    if (!pbnc_can_start_transaction(pb)) {
+        pubnub_mutex_unlock(pb->monitor);
+        return PNR_IN_PROGRESS;
+    }
+
+    rslt = pbauto_heartbeat_prepare_channels_and_ch_groups(pb, &channel, &opts.channel_group);
+    if (rslt != PNR_OK) {
+        return rslt;
+    }
+
+    pb->core.state = state;
```

Comment:
I believe this is incorrect way to set state or inconsistent way from the functionality of set_state
Please check https://github.com/pubnub/c-core/blob/master/core/pubnub_coreapi.c#L319-L408

Reply:
ahhh I see. Okay thanks! Gonna change that.